src/refiners/training_utils/batch.py

```diff
@@ -0,0 +1,194 @@
+            elif isinstance(attr_list[0], list):
+                collated_attrs[type_key] = [item for sublist in attr_list for item in sublist]
+            else:
+                raise ValueError(f"Unsupported attribute type for collation: {type(attr_list[0])}")
+
+        collated_instance = cls(**collated_attrs)
+        return collated_instance
+
+    def to(self: T, device: Device | None = None, dtype: DType | None = None) -> T:
+        for type_key in get_type_hints(self.__class__):
+            value = getattr(self, type_key)
+            if isinstance(value, Tensor):
+                setattr(self, type_key, value.to(device, dtype))
+            elif isinstance(value, list):
+                setattr(self, type_key, value)
+            else:
+                raise ValueError(f"Unsupported attribute type for to: {type(value)}")
+        return self
+
+    def __len__(self) -> int:
+        return self._length
+
+    def to_dict(self) -> dict[str, AttrType]:
+        return {type_key: getattr(self, type_key) for type_key in get_type_hints(self.__class__)}
+
+    def split(self: T) -> list[T]:
+        result: list[T] = []
+        l = len(self)
+        for i in range(l):
+            args = {type_key: getattr(self, type_key)[i : i + 1] for type_key in get_type_hints(self.__class__)}
+            result.append(self.__class__(**args))
+        return result
+
+    def __iter__(self):
+        for batch in self.split():
+            yield batch
+
+    @classmethod
+    def load(cls: Type[T], filename: Path, map_location: Device | None = None) -> T:
+        return cls(**torch_load(filename, map_location=map_location))
+
+    def save(self, filename: Path) -> None:
+        torch_save(self.to_dict(), filename)
+
+    def clone(self: T) -> T:
+        return self.__class__(**self.to_dict())
+
+    def __eq__(self, __value: object) -> bool:
+        if not isinstance(__value, self.__class__):
+            return False
+        dict = self.to_dict()
+        other_dict = __value.to_dict()
+
+        for key in dict:
+            if isinstance(dict[key], Tensor):
+                condition = dict[key] == other_dict[key]
+                casted = cast(Tensor, condition)
+                if not casted.all():
+                    return False
+            else:
+                if dict[key] != other_dict[key]:
+                    return False
+        return True
+
+    def __neq__(self, __value: object) -> bool:
+        return not self.__eq__(__value)
+
+    def __repr__(self) -> str:
+        attr_strs: list[str] = []
+        for type_key in get_type_hints(self.__class__):
+            attr_value = getattr(self, type_key)
+            if isinstance(attr_value, list):
+                attr_strs.append(f"{type_key}={attr_value}")
+            else:
+                attr_strs.append(f"{type_key}={summarize_tensor(attr_value)}")
+        return f"{self.__class__.__name__}({len(self)})[{','.join(attr_strs)}]"
```

Comment:
Let's maybe be explicit about what is in the repr
```suggestion
        return f"{self.__class__.__name__}(size={len(self)})[{','.join(attr_strs)}]"
```